userspace/libscap/scap_savefile.c

```diff
@@ -1580,15 +1732,37 @@ static int32_t scap_read_proclist(scap_t *handle, gzFile f, uint32_t block_lengt
 		{
 			handle->m_proc_callback(handle->m_proc_callback_context, handle, tinfo.tid, &tinfo, NULL);
 		}
+
+		if(sub_len && subreadsize != sub_len)
+		{
+			if(subreadsize < sub_len)

```

Comment:
I believe this, and all the other newly added occurrences, should be `>`, not `<`.

Reply:
you're right, I misunderstood the comment. It's fixed now